### ISSUE TITLE
runtime: unusual code in proc.go could use documentation

### ISSUE THREAD

**Issue by velovix** [golang/go#20717]
### What version of Go are you using (`go version`)?
go version devel +d293c85e39 Sat Jun 17 08:17:14 2017 +0000 linux/amd64

### What operating system and processor architecture are you using (`go env`)?
```GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/velovix/go"
GORACE=""
GOROOT="/home/velovix/go-source"
GOTOOLDIR="/home/velovix/go-source/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build407292918=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

Please see this discussion on golang-nuts:
https://groups.google.com/d/topic/golang-nuts/FnF3oZeJ7aY/discussion

There is some code at lines 198-202 in `/src/runtime/proc.go` that could do with a bit of documentation. The code in question can be found [here](https://github.com/golang/go/blob/df0892cbf854e76fae6e875043b05c194e37f52d/src/runtime/proc.go#L208) on Github.

Here is my current draft:
```
// In case exit fails to stop the program, try to stop it by causing a
// panic. If that doesn't work, stall the program.
// This is here to help catch problems with new ports, where exit may not
// be properly implemented.
```

I will send in a CL if that looks okay.

**Comment by ALTree** [comment]
> I will send in a CL if that looks okay.

Just sending the CL is fine: it will be reviewed (and commented on) on gerrit. Remember to link this issue in the commit message.

**Comment by robpike** [comment]
Comment I added to CL:

I've changed my mind. This doesn't need careful documentation. It's a common pattern for code in runtimes - and other systems - to cause deliberate crashes when things are clearly not working.

To someone reading the code, it should be very clear that this code will cause a trap, and if the reader is not accustomed to reading runtimes, a glance to the line immediately preceding should make it very clear why we're trapping.
